mainwindow.cpp

```diff
@@ -174,32 +176,50 @@ void MainWindow::printReport(QString city)
 }
 
 
-void MainWindow::on_pushButton_KORD_clicked()
+void MainWindow::on_pushButton_1_clicked()
+{
+    printReport(ui->pushButton_1->text());
+}
+
+void MainWindow::on_pushButton_2_clicked()
 {
-    printReport("KORD");
+    printReport(ui->pushButton_2->text());
 }
 
-void MainWindow::on_pushButton_KMDW_clicked()
+void MainWindow::on_pushButton_3_clicked()
 {
-    printReport("KMDW");
+    printReport(ui->pushButton_3->text());
 }
 
-void MainWindow::on_pushButton_KJFK_clicked()
+void MainWindow::on_pushButton_4_clicked()
 {
-    printReport("KJFK");
+    printReport(ui->pushButton_4->text());
 }
 
-void MainWindow::on_pushButton_KGUS_clicked()
+void MainWindow::on_pushButton_5_clicked()
 {
-    printReport("KGUS");
+    printReport(ui->pushButton_5->text());
 }
 
-void MainWindow::on_pushButton_KMSP_clicked()
+void MainWindow::on_pushButton_6_clicked()
 {
-    printReport("KMSP");
+    printReport(ui->pushButton_6->text());
 }
 
-void MainWindow::on_pushButton_KCMX_clicked()
+void MainWindow::on_actionSet_Buttons_triggered()
 {
-    printReport("KCMX");
+    buttonConfig *buttons = new buttonConfig;
```

Comment:
you should tell buttonConfig to be parented to the main Window:
`buttonConfig *button = new buttonConfig(this);`

Reply:
That explains something I had trouble with.